fieldmanager.php

```diff
@@ -98,6 +98,7 @@ function fieldmanager_load_file( $file ) {
 }
 
 // Load utility classes with helper functions. These can't be autoloaded.
+fieldmanager_load_file( 'libraries/parsedown/Parsedown.php' );
```

Comment:
If a site is already using Parsedown, this will throw an exception and potentially take the site down. We'll need to namespace it. Further, it should be able to be autoloaded since it's a class.

Reply:
If it's autoloaded, does it also need to be namespaced? (I get that they're not mutually exclusive on a technical level; more wondering whether autoloading is sufficient due diligence.)

Reply:
good question. two whatifs come to mind:

1. A page could theoretically be loading Parsedown very late in the request, e.g. for something in the admin footer. if this loaded first using the autoloader, it could still break that integration. I think.
2. If a site already uses Parsedown, it might be using a different version. if ours loaded first, that could produce unexpected results in the other integration; if theirs loaded first, it could produce unexpected results in Fieldmanager.

If my hunches are right, I suspect it would not be sufficient due diligence. But I'm very prepared to be wrong 😉 .

Reply:
> A page could theoretically be loading Parsedown very late in the request, e.g. for something in the admin footer. if this loaded first using the autoloader, it could still break that integration. I think.

True, if Parsedown was being loaded directly via `require` or `include` without first checking whether the class was defined. In that case, though, the developer would be at risk of the same issue from any number of libraries that might include Parsedown as a dependency.

> If a site already uses Parsedown, it might be using a different version. if ours loaded first, that could produce unexpected results in the other integration; if theirs loaded first, it could produce unexpected results in Fieldmanager.

Sold!

Reply:
I've implemented autoloading and namespaced to `Fieldmanager_Parsedown`.

The conditional is a bit clunky (see below), but I'm open to suggestions on how it should be structured. I added an explanation/call for refactoring should someone need a broader solution/autoload pattern.

Normally I'd namespace and autoload based on the path (`Fieldmanager/Libraries/Parsedown` would look in `/php/libraries/parsedown/class-parsedown.php`), but it's not a pattern anywhere else in FM, and I wanted to stay consistent.

```
/**
 * Autoload support for Parsedown.
 *
 * This is done specifically for Parsedown because it's a third party
 * library, which presents a different folder structure than other
 * autoloads. If other libraries are introduced in the future, please
 * abstract/refecator/generalize this functionality as needed.
 */
if ( 'Fieldmanager_Parsedown' === $class ) {
	return fieldmanager_load_file( 'libraries/parsedown/class-fieldmanager-parsedown.php' );
}
```